tools/jdbc/src/main/java/org/duckdb/DuckDBResultSetMetaData.java

```diff
@@ -217,11 +217,23 @@ public int getColumnDisplaySize(int column) throws SQLException {
 	}
 
 	public int getPrecision(int column) throws SQLException {
-		return 0;
+		DuckDBColumnTypeMetaData typeMetaData = typeMetadataForColumn(column);
+
+		if (typeMetaData == null) {
+			return 0;
+		}
+
+		return typeMetaData.width;
 	}
 
 	public int getScale(int column) throws SQLException {
-		return 0;
+		DuckDBColumnTypeMetaData typeMetaData = typeMetadataForColumn(column);
+
+		if (typeMetaData == null) {
```

Comment:
It appears to me that only decimal type will have this information given the code here: https://github.com/duckdb/duckdb/blob/3714e4fecc861333711525fa4ecdfc3f11daa802/tools/jdbc/src/jni/duckdb_java.cpp#L392-L426

I can add additional check that only Decimal types will even check for a typeMetadata, would that address your concerns? Any additional type metadata could be added in the future and would also need to update this check to ensure they are in sync.

Reply:
My thought was, that if for any reason other types also use the DuckDBColumnTypeMetaData in the future (which then most probably have to be extended), the null check is not sufficient anymore. A simple check if the type is decimal would prevent this and can be extended to any other type supporting scale/precision.

Reply:
Assuming that we have a convention for p/s for arbitrary types (it used to be just 0!), I don't see this as a problem.  In any case, if we want to check for inappropriate references to the values, it would be better to use an accessor that does the check. Then we could catch all bad usages, not just these ones?

Reply:
After reviewing the DuckDBColumnTypeMetaData, I would say you can leave it as it is.

Great that you fixed these two missing pieces ...

Reply:
Thanks for the review and discussion!